### PR TITLE
fix: Add rolling users validation for oncall shift API

### PR DESCRIPTION
**Pull request by ravishankar15**
# What this PR does
Adds validation for rolling users param in the shift api

## Which issue(s) this PR closes
Closes [5041](https://github.com/grafana/oncall/issues/5041)

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

**Review by joeyorlando — APPROVED** [review]
LGTM 👍 thanks for patch @ravishankar15! 🙂

File: engine/common/api_helpers/custom_fields.py [hunk 1]
```diff
@@ -102,7 +103,16 @@ def to_internal_value(self, data):
...
             return None
 
-        return queryset.filter(organization=request.user.organization, public_primary_key__in=data).distinct()
+        users = queryset.filter(organization=request.user.organization, public_primary_key__in=data).distinct()
+        users_ppk = [u.public_primary_key for u in users]
+
+        if not self.db_verification:
+            return users
+
+        for d in data:
```
Comment (matiasb) [inline engine/common/api_helpers/custom_fields.py:112]: Would this check be equivalent to do something like
`len(set(data)) == users.count()`
(ie. make sure we have a user per provided ppk)? I guess you could do a set difference to get the missing users too?

File: engine/apps/api/serializers/on_call_shifts.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ class OnCallShiftSerializer(EagerLoadingMixin, serializers.ModelSerializer):
         allow_null=True,
         required=False,
         child=UsersFilteredByOrganizationField(
-            queryset=User.objects, required=False, allow_null=True
+            queryset=User.objects, db_verification=True, required=False, allow_null=True
```
Comment (matiasb) [inline engine/apps/api/serializers/on_call_shifts.py:32]: I would suggest a different name for the param, based on my understanding of the check (see also the comment below), eg. `require_all_exist`?

**Review by matiasb — COMMENTED** [review]
Makes sense to me :+1: 
Minor suggestions added.

File: engine/common/api_helpers/custom_fields.py (hunk 1 above)
Comment (ravishankar15) [reply engine/common/api_helpers/custom_fields.py]: I have Made the changes accordingly

**Pull request merged**